server/ntb/publish/ntb_nitf.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from superdesk.metadata.item import ITEM_TYPE, CONTENT_TYPE
+from superdesk.publish.formatters.nitf_formatter import NITFFormatter
+from xml.etree.ElementTree import SubElement
+import logging
+import re
+logger = logging.getLogger(__name__)
+
+EMBED_RE = re.compile(r"<!-- EMBED START (([a-zA-Z]+) {id: \"(?P<id>.+?)\"}) -->(.*)"
+                      r"<!-- EMBED END \1 -->", re.DOTALL)
+
+
+class NTBNITFFormatter(NITFFormatter):
+
+    def can_format(self, format_type, article):
+        """
+        Method check if the article can be formatted to NTB NIT
+        :param str format_type:
+        :param dict article:
+        :return: True if article can formatted else False
+        """
+        return format_type == 'ntbnitf' and article[ITEM_TYPE] == CONTENT_TYPE.TEXT
+
+    def _format_tobject(self, article, head):
+        category = ''
+        for s in article.get('subject', []):
+            if s.get('scheme') == 'category':
+                category = s['qcode']
+        return SubElement(head, 'tobject', {'tobject.type': category})
+
+    def _format_docdata(self, article, docdata):
+        super()._format_docdata(article, docdata)
+        if 'slugline' in article:
+            SubElement(docdata, 'du-key', attrib={'version': '1', 'key': article['slugline']})
+
+    def _format_subjects(self, article, tobject):
+        subjects = [s for s in article.get('subject', []) if s.get("scheme") == "subject_custom"]
```

Comment:
Subject_custom is a scheme added for NTB only. We have to avoid to use custom fields in code.